### PR TITLE
make the ipc listen address configurable

### PR DESCRIPTION
**Pull request by arbll**
### What does this PR do?

Makes the ipc listen address configurable.

### Motivation

Some setups might resolve localhost to something different than `127.0.0.1` / `::1`.

### CODE REVIEW

**Comment by ogaca-dd** [comment]
What do you think about creating a function that share the common logic for bind_ipc and cmd_port?

**Review by remeh — DISMISSED** [review]
That LGTM.

**Comment by arbll** [comment]
@ogaca-dd You mean creating a function returning `<address>:<port>` ? I don't have a strong opinion on this since it offers the same readability IMO. Maybe I prefer slightly the current one as it is a bit more flexible.

File: pkg/flare/cluster_checks.go [hunk 1]
```diff
@@ -22,7 +22,11 @@ import (
 
 // GetClusterChecks dumps the clustercheck dispatching state to the writer
 func GetClusterChecks(w io.Writer) error {
-	urlstr := fmt.Sprintf("https://localhost:%v/api/v1/clusterchecks", config.Datadog.GetInt("cluster_agent.cmd_port"))
+	ipcAddress, err := config.GetIPCAddress()
+	if err != nil {
+		return err
+	}
+	urlstr := fmt.Sprintf("https://%v:%v/api/v1/clusterchecks", ipcAddress, config.Datadog.GetInt("cluster_agent.cmd_port"))
```
Comment (olivielpeau) [inline pkg/flare/cluster_checks.go]: I think the address here is supposed to be the address of the IPC of the cluster agent, not of the node agent, see https://github.com/DataDog/datadog-agent/blob/4a6a8c9788ea868bdc88fb24a8694d1a7c829de3/cmd/cluster-agent/app/clusterchecks.go#L48.

So probably best to leave this to `localhost`.

File: pkg/flare/config_check.go [hunk 2]
```diff
@@ -91,7 +94,11 @@ func GetConfigCheck(w io.Writer, withDebug bool) error {
 
 // GetClusterAgentConfigCheck proxies GetConfigCheck overidding the URL
 func GetClusterAgentConfigCheck(w io.Writer, withDebug bool) error {
-	configCheckURL = fmt.Sprintf("https://localhost:%v/config-check", config.Datadog.GetInt("cluster_agent.cmd_port"))
+	ipcAddress, err := config.GetIPCAddress()
+	if err != nil {
+		return err
+	}
+	configCheckURL = fmt.Sprintf("https://%v:%v/config-check", ipcAddress, config.Datadog.GetInt("cluster_agent.cmd_port"))
```
Comment (olivielpeau) [inline pkg/flare/config_check.go]: same here

**Review by olivielpeau — COMMENTED** [review]
one comment on the DCA-specific code, other than that LGTM

File: pkg/flare/config_check.go [hunk 3]
```diff
@@ -91,7 +94,7 @@ func GetConfigCheck(w io.Writer, withDebug bool) error {
 
 // GetClusterAgentConfigCheck proxies GetConfigCheck overidding the URL
 func GetClusterAgentConfigCheck(w io.Writer, withDebug bool) error {
-	configCheckURL = fmt.Sprintf("https://localhost:%v/config-check", config.Datadog.GetInt("cluster_agent.cmd_port"))
+	configCheckURL = fmt.Sprintf("https://%v:%v/config-check", config.Datadog.GetString("bind_ipc"), config.Datadog.GetInt("cluster_agent.cmd_port"))
```
Comment (olivielpeau) [inline pkg/flare/config_check.go]: one more here 😬

**Review by olivielpeau — APPROVED** [review]
🚢

**Pull request merged**